vimdoc/module.py

```diff
@@ -141,25 +142,35 @@ def Close(self):
     if ((self.GetCollection(vimdoc.FLAG) or
          self.GetCollection(vimdoc.SETTING)) and
         'config' not in self.sections):
-      config = Block()
-      config.SetType(vimdoc.SECTION)
+      config = Block(vimdoc.SECTION)
       config.Local(id='config', name='Configuration')
       self.Merge(config)
-    if not self.order:
-      self.order = []
-      for builtin in [
-          'intro',
-          'config',
-          'commands',
-          'autocmds',
-          'settings',
-          'dicts',
-          'functions',
-          'exceptions',
-          'mappings',
-          'about']:
-        if builtin in self.sections or builtin in self.backmatters:
-          self.order.append(builtin)
+
+    # Use explicit order as partial ordering and merge with default section
+    # ordering. All custom sections must be ordered explicitly.
+    # Custom sections will be ordered after all sections besides 'about'.
+    order = self.order or []
+    builtins_pre = []
+    for builtin in [
+        'intro',
+        'config',
+        'commands',
+        'autocmds',
+        'settings',
+        'dicts',
+        'functions',
+        'exceptions',
+        'mappings']:
+      if builtin in self.sections or builtin in self.backmatters:
+        if builtin not in order:
+          builtins_pre.append(builtin)
+    order = builtins_pre + order
```

Comment:
I think it's reasonable to stick the user-specified parts in the middle of the list for now, though perhaps it'd be worth taking a look at how this is used in practice - I can certainly see the attraction of the more complicated ordering.

However, I'm getting a little bit confused here. `backmatters` must have a subset of the keys of `sections`, I think, otherwise we fail just below, at L163-165.

However, we then set `known = set(self.sections) | set(self.backmatters)`, which must surely just be the same as `set(self.sections)`, unless I'm missing something?

Either way, if we pull up L163-166 to before this, we can perhaps write this more simply (WDYT?):

```python
DEFAULT_ORDER_PRE = [
  'intro',
  'config',
  'commands',
  'autocmds',
  'settings',
  'dicts',
  'functions',
  'exceptions',
  'mappings']
DEFAULT_ORDER_POST = [
  'about']
```
```python
for backmatter in self.backmatters:
  if backmatter not in self.sections:
    raise error.NoSuchSection(backmatter)
known = set(self.sections)  # I think?

explicit_order = self.order or []
order_pre = [section for section in DEFAULT_ORDER_PRE
             if section in known and section not in explicit_order]
order_post = [section for section in DEFAULT_ORDER_POST
              if section in known and section not in explicit_order]
order = order_pre + explicit_order + order_post
```

(Incidentally, what indent are we aiming for, PEP8 or Google? We seem to use both.)